demonstrations/tutorial_implicit_diff_susceptibility.py

```diff
@@ -0,0 +1,656 @@
+differentiation through the full optimization that finds :math:`z^{*}(a)`
+would require keeping track of all intermediate variables and steps in the optimization
+and differentiating through them. This could quickly become computationally
+expensive and memory intensive for quantum algorithms. Implicit differentiation
+provides an alternative way to efficiently compute such a gradient. 
+
+:math:`\partial_{a} \langle A\rangle` is the so-called 
+*generalized susceptibility* arising from condensed-matter physics. The computation
+of this quantity would otherwise require tedious analytical derivations or
+finite difference approximations. Similarly there exist various other
+interesting quantities written as gradients of a ground-state solution, 
+e.g., nuclear forces in quantum chemistry, permanent electric dipolestatic
+polarizability, the static hyperpolarizabilities of various orders,
+fidelity susceptibilities, and geometric tensors. All such
+quantities could possibly be computed using implicit differentiation on quantum
+devices. In our recent work, we present a unified way to implement such
+computations and other applications of implicit differentiation through
+variational quantum algorithms [#Ahmed2022]_.
+
+In this demo, we will show how to compute implicit gradients through a variational
+algorithm written in PennyLane using a modular implicit differentiation
+implementation provided by the tool JAXOpt [#Blondel2021]. We compute the generalized
+susceptibility for a spin system by using a variational ansatz to compute a
+ground-state and implicitly differentiating through it. In order to compare
+the implicit solution, we will find the exact ground-state through eigendecomposition
+and take gradients through the eigendecomposition using automatic differentiation.
+Even though for the small number of spins we consider here, eigendecomposition and
+gradient computation through it suffice, for larger systems it quickly becomes
+infeasible.
+
+
+Implicit Differentiation
+------------------------
+
+We consider differentiating a solution of the root-finding problem defined by
+
+.. math::
+
+    f(z, a) = 0.
+
+A function :math:`z^{*}(a)` that satisfies :math:`f(z^{*}(a), a) = 0` gives a
+solution map for fixed values of :math:`a`. An explicit analytical solution
+is however difficult to obtain in general. Therefore differentiating
+:math:`\partial_a z^{*}(a)` is not always possible directly. However, some iterative
+algorithms could compute the solution starting from an initial set of values
+for :math:`z`, e.g., using a fixed-point solver. The optimality condition 
+:math:`f(z^{*}(a), a) = 0` tells the solver when a solution is found.
+A brute-force application of automatic differentiation through
+the full iterative solver would however require keeping track of all the steps
+in solving the fixed-point problem. This can quickly make computing gradients
+memory intensive.
```

Comment:
```suggestion
A function :math:`z^{*}(a)` that satisfies :math:`f(z^{*}(a), a) = 0` gives a
solution map for fixed values of :math:`a`. An explicit analytical solution
is, however, difficult to obtain in general. This means that the direct differentiation of
:math:`\partial_a z^{*}(a)` is not always possible. Despite that, some iterative
algorithms may be able to compute the solution by starting from an initial set of values
for :math:`z`, e.g., using a fixed-point solver. The optimality condition 
:math:`f(z^{*}(a), a) = 0` tells the solver when a solution is found.
```
Can't have 'however' in every sentence.
Also the last two sentences were copy-pasted from a previous paragraph, so I removed them.